### ISSUE TITLE
Performance Issues - Time to mount cache

### ISSUE THREAD

**Issue by asleepingpanda** [apollographql/apollo-cache-persist#336]
I have a React website that I'm calling 

```
persistCache({
                cache,
                storage: window.localStorage,
            }).then(() => {
                client.onResetStore(async () => cache.writeData({ data: initData }));
                setClient(client);
            });
```

This process takes anywhere from 4-5 seconds on average for the first load. Every subsequent load time is near instantaneous, although I would really like to get the initial load time to be a bit more performant. 

I'm using:
React: 16.8.6
React DOM: 16.8.6
React Apollos: 3.1.1
Apollo Cache Persist: 0.1.1

I'm using React Hooks and followed the documentation seen on the README to get this set up. Again, the performance overall is fantastic, although the very first time a user loads the site, it's not a great experience and I'm just not sure what to do to optimize this performance.

**Comment by wtrocki** [comment]
How you measure the time to load it. In inspector what is consuming the most of time - JSON.parse? What is the size of your cache?

**Comment by asleepingpanda** [comment]
> How you measure the time to load it
I did a speed test without attaching Local Storage and another with it. 

> You mean navigation to another page?
Yes. After the local storage has successfully attached the first time (which makes page load time anywhere from 4-5 seconds), every refresh or page navigation is nearly instantaneous with load times being as low as under a second. Obviously, this performance is fantastic and exactly what we're looking for. But the initial load time is something that I would love to optimize if possible. If nothing else, even getting a loading indicator to show would suffice for the time being.

Following the documentation on the README, I was unable to get the first load of my web app to get under 4 seconds. I also wasn't able to get a loading indicator to show during that 4-5 second period. I'm not sure if this info is very helpful or if there's anything that can be done, but I figured I would reach out and ask! I'd be more than happy to share the repo if that would also help identify potential fixes
